### DISCUSSION TITLE
Can Storybook (docs) show source including import statements and function definitions?

### DISCUSSION THREAD

**Discussion by nse7ens** [storybookjs/storybook#15542] [Help]
I'm using Storybook to show my TypeScript React components, using MDX and CSF to render the stories.
So far everything works great, but I was still wondering:

Is there a way to have stories displayed as "real usage" full code?
E.g.: including import statements, function definitions, etc etc?

For example:

I have a story now describing a (controlled) component. For this example, I show a working sample including handling the controlled props.
To show how this works, I configured the addon-docs part to have `source: { type: "code" }`. This shows the story as defined:

```
(
	{ selected: selectedProp, ...args }: Partial<ITreeViewProps>,
	context: StoryContext
) => {
	const [selected, setSelected] = React.useState<string | null>(null);

	// Changes selection when prop changes
	React.useEffect(() => {
		setSelected(selectedProp!);
	}, [selectedProp]);

	// Callback for managing controlled selection state
	const onNodeSelect = (event: React.ChangeEvent<{}>, nodeId?: string): void => {
		setSelected(nodeId!);
	};

	return (
		<div>
			<TreeView {...args} selected={selected} onNodeSelect={onNodeSelect}>
				<TreeItem nodeId={"Parent 1"} label={"Parent 1"}>
					<TreeItem nodeId={"Child 1"} label={"Child 1"} />
				</TreeItem>
				<TreeItem nodeId={"Parent 2"} label={"Parent 2"} />
			</TreeView>
		</div>
	);
}
```
What I would like, is that I can show an "Example" component, complete with all the boilerplate, so that it really shows a "complete" usable sample.
E.g:
```
import * as React from "react";
import { TreeView, TreeItem } from "../Tree";

export const Sample: React.FC<{}> = (props) => {
	const { selected: selectedProp } = props;
	const [selected, setSelected] = React.useState<string | null>(null);

	// Changes selection when prop changes
	React.useEffect(() => {
		setSelected(selectedProp!);
	}, [selectedProp]);

	// Callback for managing controlled selection state
	const onNodeSelect = (event: React.ChangeEvent<{}>, nodeId?: string): void => {
		setSelected(nodeId!);
	};

	return (
		<div>
			<TreeView {...args} selected={selected} onNodeSelect={onNodeSelect}>
				<TreeItem nodeId={"Parent 1"} label={"Parent 1"}>
					<TreeItem nodeId={"Child 1"} label={"Child 1"} />
				</TreeItem>
				<TreeItem nodeId={"Parent 2"} label={"Parent 2"} />
			</TreeView>
		</div>
	);
}
```

**Comment by eddieshowcase** [comment]
I also have the same issue. I know there is this option documented on README: https://github.com/storybookjs/storybook/tree/next/addons/storysource#displaying-full-source, but doesn't seem to work for me or show full source. I'm using version 6.3.4.

**Comment by romakulakov** [comment]
I think there is no automated way to achieve this. But you could use any string as a source code, more info https://storybook.js.org/docs/react/writing-docs/doc-blocks#docspage-1
